doc/developer/platform/reconciliation.md

```diff
@@ -0,0 +1,59 @@
+# Reconciliation of the ADAPTER/COMPUTE communication
+
+The current proposal for Materialize's platform includes the separation of the ADAPTER and COMPUTE components into distinct processes.
+Each represents its own failure domain and hence we need to design the communication protocols around failures and active replication (AR).
+In general, we need to define how commands and responses can be reconciled.
+
+## Command reconciliation
+
+Commands represent instructions the ADAPTER sends to COMPUTE instances.
+* The ADAPTER may restart independently of a COMPUTE instance.
+* Reconciling commands is the mechanism to match a COMPUTE instance's state to what the ADAPTER expects after it starting.
+* On startup, the ADAPTER provides the current configuration to the COMPUTE instances.
+* COMPUTE instances can match the configuration to their own state and only apply changes.
```

Comment:
Can we future-proof this by saying "On startup, the COMPUTE `Controller` provides ..."? I think the work here is between the controller and instances, rather than ADAPTER and COMPUTE.